### PR TITLE
CSD fix

### PR DESCRIPTION
**Pull request by shipyy**
* fix for #349
* removed 2 of the colours that seemed not to work (I might be negative IQ so idk)
* added a few more colours
* fixed color on the center hud speed , it only seemed to display the correct colors when using "momentum" option

### CODE REVIEW

**Comment by tslashd** [comment]
I did some quick testing and it looks and behaves good. 
Only thing I would suggest would be to add this for ease of use:

`RegConsoleCmd("sm_csd", Command_CenterSpeed, "[surftimer] [settings] on/off - toggle center speed display");`

File: addons/sourcemod/scripting/surftimer/db/queries.sp [hunk 1]
```diff
@@ -39,7 +39,7 @@ char sql_insertmaptier[] = "INSERT INTO ck_maptier (mapname, tier) VALUES ('%s',
...
...DEFAULT '', `timer` int(11) NOT NULL DEFAULT '1', `hide` int(11) NOT NULL DEFAULT '0', `sounds` int(11) NOT NULL DEFAULT '1', `chat` int(11) NOT NULL DEFAULT '0', `viewmodel` int(11) NOT NULL DEFAULT '1', `autobhop` int(11) NOT NULL DEFAULT '1', `checkpoints` int(11) NOT NULL DEFAULT '1', `gradient` int(11) NOT NULL DEFAULT '0', `speedmode` int(11) NOT NULL DEFAULT '0', `centrespeed` int(11) NOT NULL DEFAULT '0', `centrehud` int(11) NOT NULL DEFAULT '1', teleside int(11) NOT NULL DEFAULT '0', `module1c` int(11) NOT NULL DEFAULT '1', `module2c` int(11) NOT NULL DEFAULT '2', `module3c` int(11) NOT NULL DEFAULT '3', `module4c` int(11) NOT NULL DEFAULT '4', `module5c` int(11) NOT NULL DEFAULT '5', `module6c` int(11) NOT NULL DEFAULT '6', `sidehud` int(11) NOT NULL DEFAULT '1', `module1s` int(11) NOT NULL DEFAULT '5', `module2s` int(11) NOT NULL DEFAULT '0', `module3s` int(11) NOT NULL DEFAULT '0', `module4s` int(11) NOT NULL DEFAULT '0', `module5s` int(11) NOT NULL DEFAULT '0', prestrafe int(11) NOT NULL DEFAULT '0', cpmessages int(11) NOT NULL DEFAULT '1', wrcpmessages int(11) NOT NULL DEFAULT '1', hints int(11) NOT NULL DEFAULT '1', PRIMARY KEY (`steamid`)) DEFAULT CHARSET=utf8mb4;";
```
Comment (Bara) [inline addons/sourcemod/scripting/surftimer/db/queries.sp:42]: I'm not a fan of changing a default value without a query which set the same value of exists tables.

File: addons/sourcemod/scripting/surftimer/commands.sp [hunk 2]
```diff
@@ -236,15 +237,17 @@ public Action Command_ToggleNcTriggers(int client, int args) {
...
 		CPrintToChat(client, "%t", "CenterSpeedOff", g_szChatPrefix);
-	} else {
+	}
+	else{
 		g_bCenterSpeedDisplay[client] = true;
-		SetHudTextParams(-1.0, 0.30, 1.0, 255, 255, 255, 255, 0, 0.25, 0.0, 0.0);
-		CreateTimer(0.1, CenterSpeedDisplayTimer, client, TIMER_FLAG_NO_MAPCHANGE|TIMER_REPEAT);
 		CPrintToChat(client, "%t", "CenterSpeedOn", g_szChatPrefix);
 	}
+
```
Comment (Bara) [inline addons/sourcemod/scripting/surftimer/commands.sp:248]: 2 empty lines? just for the optic :P

**Review by Bara — CHANGES_REQUESTED** [review]

**Comment by Bara** [comment]
quick review, would be nice if you can do this. otherwise it's nice to some activity again ;)

File: addons/sourcemod/scripting/surftimer/db/queries.sp (hunk 1 above)
Comment (shipyy) [reply addons/sourcemod/scripting/surftimer/db/queries.sp:42]: i changed it so it uses "white" gradient as default, should i change it to the previous value?

File: addons/sourcemod/scripting/surftimer/hooks.sp [hunk 3]
```diff
@@ -824,6 +818,9 @@ public Action OnLogAction(Handle source, Identity ident, int client, int target,
 
 public Action OnPlayerRunCmd(int client, int &buttons, int &impulse, float vel[3], float angles[3], int &weapon, int &subtype, int &cmdnum, int &tickcount, int &seed, int mouse[2])
 {
+	//calling this function here makes it refresh musch faster/smoother
```
[the remark this answers is not in the document]
Comment (shipyy) [reply addons/sourcemod/scripting/surftimer/hooks.sp:821]: im confusion, dont understand what u mean by this xD

[the remark this answers is not in the document]
Comment (DaLLuZZ) [reply addons/sourcemod/scripting/surftimer/hooks.sp:821]: > it's "faster", but when you type (mouse moving will not handled by OnPlayerRunCmd) nothing it shows nothing.

even if you type nothing, this `OnPlayerRunCmd` is processing every frame, just test it

[the remark this answers is not in the document]
Comment (Bara) [reply addons/sourcemod/scripting/surftimer/hooks.sp:821]: > > it's "faster", but when you type (mouse moving will not handled by OnPlayerRunCmd) nothing it shows nothing.
> 
> even if you type nothing, this `OnPlayerRunCmd` is processing every frame, just test it

Yes, but forward will not called when no Buttons will pressed.

**Comment by shipyy** [comment]
sent commits fixing reported issues, i did end up reverting default values of speedgradient back to "3" btw

File: addons/sourcemod/scripting/surftimer/hooks.sp (hunk 3 above)
[the remark this answers is not in the document]
Comment (shipyy) [reply addons/sourcemod/scripting/surftimer/hooks.sp:821]: i understand that OnPlayerRunCmd gets called everyframe but should the CSD be display not matter what? (even if the player is not moving?)

**Comment by shipyy** [comment]
pls dont accept it, i have completly fuck my repos kms

**Comment by shipyy** [comment]
my repos went boom

**Pull request closed without merging**